typings/glamorous.d.ts

```diff
@@ -40,6 +40,12 @@ interface ThemeProps {
 
 export class ThemeProvider extends React.Component<ThemeProps, any> { }
 
+export declare function withTheme
+  <P>(
+    component:
+      | React.ComponentClass<P>
```

Comment:
Your definition of Component<T> looks like it could also be used in the current `GlamorousInterface`. Does this refactoring look correct? (I'm brand new to TypeScript, I honestly don't understand the generics or many other things yet).

```js
export type Component<T> = React.ComponentClass<T> | React.StatelessComponent<T>;

export interface GlamorousInterface extends HTMLGlamorousInterface, SVGGlamorousInterface {
  <P>(
    component:Component<P>,
    options?: GlamorousOptions
  ): StyledFunction<P, CSSProperties | React.SVGAttributes<any>>

export declare function withTheme
  <P>(component: Component<P & ThemeProps>): GlamorousComponent<P>;
```